ui/lib/core/addon/components/search-select.js

```diff
@@ -103,77 +127,104 @@ export default Component.extend({
         ...(this.idKey !== 'id' && this.customizeObject(matchingOption)),
       };
     });
-    this.set('selectedOptions', formattedOptions);
-    if (this.options) {
-      options = this.options.concat(options).uniq();
+  }
+
+  @task
+  *fetchOptions() {
+    if (this.args.parentManageSelected) {
+      this.selectedOptions = this.args.parentManageSelected;
     }
-    this.set('options', options);
-  },
-  fetchOptions: task(function* () {
-    if (!this.models) {
-      if (this.options) {
-        this.formatOptions(this.options);
+    this.dropdownOptions = [];
+
+    if (!this.args.models) {
+      if (this.args.options) {
+        const options = this.args.options;
+        if (options.some((e) => Object.keys(e).includes('groupName'))) {
+          this.dropdownOptions = options;
+        } else {
+          this.dropdownOptions = [...this.addSearchText(options)];
+          this.selectedOptions = this.formatSelectedAndUpdateDropdown(this.selectedOptions);
+        }
       }
       return;
     }
```

Comment:
It looks like this could be combined into one if statement and you could get rid of the nested if.

Reply:
Great point! I wonder why it was separated in the old component 🤔

Reply:
Oh! I think it was separated because if there are no models and no options, we want to return completely from `fetchOptions` - which wouldn't happen if this was combined to `if (!this.args.models && this.args.options)`

But there may be a cleaner way to relay this that doesn't require a nested `if` statement - by adding a return before the for loop below? What do you think?

Reply:
```suggestion
    if (!this.args.models && !this.args.options) {
        const options = this.args.options;
        if (options.some((e) => Object.keys(e).includes('groupName'))) {
          this.dropdownOptions = options;
        } else {
          this.dropdownOptions = [...this.addSearchText(options)];
          this.selectedOptions = this.formatSelectedAndUpdateDropdown(this.selectedOptions);
        }
  }
  
  if (!this.args.models) return
```

Reply:
> Oh! I think it was separated because if there are no models and no options, we want to return completely from `fetchOptions` - which wouldn't happen if this was combined to `if (!this.args.models && this.args.options)`
> 
> But there may be a cleaner way to relay this that doesn't require a nested `if` statement - by adding a return before the for loop below? What do you think @zofskeez ?

Oh ya good point I missed that! I think it's probably better to leave as is. It was a nit on my part and not worth risky a potential regression.